### PR TITLE
[CI] Add reusable workflow for test-tube tests

### PR DESCRIPTION
**Pull request by lubkoll**

### CODE REVIEW

File: smart-contracts/contracts/dex-router-osmosis/.cargo/config.toml [hunk 1]
```diff
@@ -2,5 +2,5 @@
 wasm = "build --release --target wasm32-unknown-unknown"
 unit-test = "test --lib"
 schema = "run --bin schema"
-test-tube = "test --package dex-router-osmosis --lib -- --include-ignored test_tube:: --nocapture --test-threads=1"
```
Comment (magiodev) [inline smart-contracts/contracts/dex-router-osmosis/.cargo/config.toml:5]: `--test-threads=1`  is needed due to some race condition that was causing not deterministic test outcomes.

File: smart-contracts/justfile [hunk 2]
```diff
@@ -34,4 +34,11 @@ workspace-optimize:
             --mount type=volume,source="$(basename "$(pwd)")_cache",target=/target \
             --mount type=volume,source=registry_cache,target=/usr/local/cargo/registry \
             --platform linux/amd64 \
-            cosmwasm/workspace-optimizer:0.16.0; fi
+            cosmwasm/optimizer:0.16.0; fi
+
+optimize-single-contract:
```
Comment (magiodev) [inline smart-contracts/justfile]: I would call that `contract-optimize` to maintain some consistency with the above command.

**Review by magiodev — CHANGES_REQUESTED** [review]

File: smart-contracts/contracts/dex-router-osmosis/.cargo/config.toml (hunk 1 above)
Comment (lubkoll) [reply smart-contracts/contracts/dex-router-osmosis/.cargo/config.toml:5]: added back

File: smart-contracts/justfile (hunk 2 above)
Comment (lubkoll) [reply smart-contracts/justfile]: removed, that did slip through from previous tests.
While that is convenient, we don't need it here

**Review by magiodev — APPROVED** [review]

**Pull request merged**